app/controllers/session_controller.rb

```diff
@@ -0,0 +1,33 @@
+# require './config/environment'
+
+class SessionController < ApplicationController
+
+  get '/sessions/new' do
+    erb :"sessions/new.html"
+  end
+
+  post '/sessions' do
+    user = User.find_by(email: params[:email])
+
+
+    if user&.authenticate(params[:password])
+      session[:loggedin] = params[:email]
+      flash[:notice] = "You are logged in as #{user.email}"
+      redirect "/"
+    else
+      @email = params[:email]
```

Comment:
Rather than storing the email as instance variable, you can pass it into the `erb` template via. `locals`